File: 2350-find-closest-number-to-zero/find-closest-number-to-zero.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var findClosestNumber = function (nums) {
    closet = 0

    for (let i = 1; i < nums.length; i++) {
        console.log(closet)
        if (Math.abs(nums[i]) == Math.abs(nums[closet])) {
            if (nums[i] > nums[closet]){
                closet = i
            }
        }
        else if(Math.abs(nums[i]) < Math.abs(nums[closet])) {
            closet = i
        }
    }

    return nums[closet]
};

File: 2350-find-closest-number-to-zero/find-closest-number-to-zero.go
func Abs(x int) int {
	if x < 0 {
		return -x
	}
	return x
}

func findClosestNumber(nums []int) int {
	closest := 0

	for i := 1; i < len(nums); i++ {
		if Abs(nums[i]) <= Abs(nums[closest]) {
            if Abs(nums[i]) == Abs(nums[closest]){
                if(nums[i] > nums[closest]){
                    closest = i
                }else{
                    continue
                }
            }
            closest = i
		}
	}

	return nums[closest]
}